### PR TITLE
maintainers

### PR DESCRIPTION
**Pull request by hhazewinkel**
Add maintainers.isRsMaintainer()
Add maintainers.isEnduserMaintainer()
Cleanup and add 'verify' Maintainers interactions.

### CODE REVIEW

File: whois-commons/src/main/java/net/ripe/db/whois/common/domain/Maintainers.java [hunk 1]
```diff
@@ -68,6 +68,10 @@ public Maintainers(
         return dbmMaintainers;
     }
 
+    public boolean isEnduserMaintainer(Set<CIString> mntner) {
+        return !Sets.intersection(enduserMaintainers, mntner).isEmpty();
+    }
+
     public boolean isRsMaintainer(CIString mntner) {
         return isRsMaintainer(Sets.newHashSet(mntner));
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/domain/Maintainers.java]: Collections.singleton(mntner) is faster than creating a new Set with one element

File: whois-query/src/test/java/net/ripe/db/whois/query/planner/AbuseCFinderTest.java [hunk 2]
```diff
@@ -52,6 +55,9 @@ public void getAbuseContacts_oneOrgReferenceNoAbuseC() {
         final String abuseContact = subject.getAbuseContact(inetnum);
 
         assertThat(abuseContact, is(nullValue()));
+
+        verify(maintainers).isRsMaintainer(ciSet());
```
Comment (eshryane) [inline whois-query/src/test/java/net/ripe/db/whois/query/planner/AbuseCFinderTest.java]: Should we test maintainers in this class? Can we put extra checks into MaintainerTest instead?

File: whois-update/src/test/java/net/ripe/db/whois/update/authentication/strategy/MntByAuthenticationTest.java [hunk 3]
```diff
@@ -66,6 +68,8 @@ public void supports_every_object_with_a_mntby_attribute() {
 
         when(update.getType()).thenReturn(ObjectType.INETNUM);
         assertThat(subject.supports(update), is(true));
+
+        verifyZeroInteractions(maintainers);
```
Comment (eshryane) [inline whois-update/src/test/java/net/ripe/db/whois/update/authentication/strategy/MntByAuthenticationTest.java]: move these tests to MaintainerTest ?

**Comment by eshryane** [comment]
Looks good apart from comments! A nice refactoring, much clearer.

File: whois-update/src/test/java/net/ripe/db/whois/update/authentication/strategy/MntByAuthenticationTest.java (hunk 3 above)
Comment (hhazewinkel) [inline whois-update/src/test/java/net/ripe/db/whois/update/authentication/strategy/MntByAuthenticationTest.java]: I guess, that is a different point. I.E. cleaning up or improving the organisation of tests.

File: whois-query/src/test/java/net/ripe/db/whois/query/planner/AbuseCFinderTest.java (hunk 2 above)
Comment (hhazewinkel) [inline whois-query/src/test/java/net/ripe/db/whois/query/planner/AbuseCFinderTest.java]: I guess, that is a different point. I.E. cleaning up or improving the organisation of tests.

**Pull request merged**